.github/workflows/macOS-CMake-toolchain.yml

```diff
@@ -0,0 +1,87 @@
+
+      - name: Install swift.org toolchain
+        run: |
+          curl -sOL https://swift.org/builds/development/xcode/${TOOLCHAIN_NAME}/${TOOLCHAIN_NAME}-osx.pkg
+          xattr -dr com.apple.quarantine ${TOOLCHAIN_NAME}-osx.pkg
+          # Install "source" toolchain. This toolchain is used to build tensorflow/swift-apis.
+          installer -pkg ${TOOLCHAIN_NAME}-osx.pkg -target CurrentUserHomeDirectory
+          # Install "destination" toolchain. tensorflow/swift-apis is installed into this toolchain.
+          mkdir /tmp/destination-toolchain
+          sudo installer -pkg ${TOOLCHAIN_NAME}-osx.pkg -target /tmp/destination-toolchain
+          rm -rf ${TOOLCHAIN_NAME}-osx.pkg
+
+      - name: Install pre-built TensorFlow and X10 libraries
+        run: |
+          curl -sL https://artprodeus21.artifacts.visualstudio.com/A8fd008a0-56bc-482c-ba46-67f9425510be/3133d6ab-80a8-4996-ac4f-03df25cd3224/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2NvbXBuZXJkL3Byb2plY3RJZC8zMTMzZDZhYi04MGE4LTQ5OTYtYWM0Zi0wM2RmMjVjZDMyMjQvYnVpbGRJZC80NTU3NC9hcnRpZmFjdE5hbWUvdGVuc29yZmxvdy1kYXJ3aW4teDY00/content?format=zip -o tensorflow-darwin-x64.zip
+          unzip tensorflow-darwin-x64.zip
+          mv tensorflow-darwin-x64/Library/tensorflow-2.4.0 ~/Library/
+
+      - name: Build tensorflow/swift-apis via CMake
+        run: |
+          echo $(TOOLCHAINS=org.swift.${TOOLCHAIN_SHORT_DATE}a xcrun -f swiftc)
+          cmake \
+            -B build \
+            -D BUILD_SHARED_LIBS=YES \
+            -D CMAKE_BUILD_TYPE=Release \
+            -D CMAKE_INSTALL_PREFIX=${DEST_TOOLCHAIN_LOCATION}/usr \
+            -D CMAKE_Swift_COMPILER=$(TOOLCHAINS=org.swift.${TOOLCHAIN_SHORT_DATE}a xcrun -f swiftc) \
+            -D CMAKE_OSX_DEPLOYMENT_TARGET=10.9 \
+            -D TENSORFLOW_USE_STANDARD_TOOLCHAIN=YES \
+            -D X10_LIBRARY=${HOME}/Library/tensorflow-2.4.0/usr/lib/libx10.dylib \
+            -D X10_INCLUDE_DIRS=${HOME}/Library/tensorflow-2.4.0/usr/include \
+            -D BUILD_TESTING=NO \
+            -G Ninja \
+            -S ${PWD}
+
+      - name: Install tensorflow/swift-apis into toolchain
+        run: |
+          cmake --build build --target install
+
+      - name: Build toolchain package installer
+        run: |
+          TOOLCHAIN_ROOT_LOCATION="/Library/Developer/Toolchains/${TOOLCHAIN_NAME}.xctoolchain"
+          TF_TOOLCHAIN_NAME="swift-tensorflow-DEVELOPMENT-SNAPSHOT-${TOOLCHAIN_DATE}-a"
+          pkgbuild --identifier org.tensorflow-${TOOLCHAIN_SHORT_DATE} \
+                   --install-location ${TOOLCHAIN_ROOT_LOCATION} \
+                   --version 5.0.${TOOLCHAIN_SHORT_DATE} \
+                   --root ${DEST_TOOLCHAIN_LOCATION} \
+                   ${TF_TOOLCHAIN_NAME}-osx.pkg
+
+      - name: Upload toolchain package installer
+        uses: actions/upload-artifact@v2
+        with:
+          name: swift-tensorflow-DEVELOPMENT-SNAPSHOT-${{ env.toolchain-date }}-a-osx.pkg
+          path: '*.pkg'
```

Comment:
I think that this needs to be made conditional.  This will do a GH release per PR.

Reply:
That's a good point. I wonder what do you think is a good condition, and how to add it?

Reply:
I tried adding some conditional steps, but they still seem to run unconditionally even when the guard condition is `false`. Any clue why or how to do better? 

https://github.com/tensorflow/swift-apis/blob/0ba6914526a7b65bec1da63e993804e9ff70d4b8/.github/workflows/macOS-CMake-toolchain.yml#L83-L99

https://github.com/tensorflow/swift-apis/runs/1660455199

Reply:
I suppose that we _could_ do this on tags only.  I think that tags are uncommon enough where it may make sense for building on a tag even when its not an actual release but some point that we want to archive.